File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.js
const fs = require("fs");
const input = fs
  .readFileSync(0)
  .toString()
  .trim()
  .split("\n")
  .map((el) => el.split(" ").map(Number));

const [nodeCnt, edgeCnt, start] = input.shift();

const graph = Array.from({ length: nodeCnt + 1 }, () => []);
for (let i = 0; i < edgeCnt; i++) {
  const [v1, v2] = input[i];
  graph[v1].push(v2);
  graph[v2].push(v1);
}

graph.forEach((el) => el.sort((a, b) => a - b));

const dfsVisited = Array.from({ length: nodeCnt + 1 }, () => false);
const dfsRes = [];
const dfs = (node) => {
  if (!dfsVisited[node]) {
    dfsVisited[node] = true;
    dfsRes.push(node);

    for (const next of graph[node]) {
      if (!dfsVisited[next]) dfs(next);
    }
  }
};

const bfsVisited = Array.from({ length: nodeCnt + 1 }, () => false);
const bfsRes = [];
const queue = [];
const bfs = (node) => {
  bfsVisited[node] = true;
  bfsRes.push(node);
  queue.push(node);

  while (queue.length !== 0) {
    const cur = queue.shift();

    graph[cur].forEach((v) => {
      if (!bfsVisited[v]) {
        bfsVisited[v] = true;
        bfsRes.push(v);
        queue.push(v);
      }
    });
  }
};

dfs(start);
bfs(start);

console.log(dfsRes.join(" "));
console.log(bfsRes.join(" "));

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
from collections import defaultdict, deque

node_cnt, edge_cnt, start = list(map(int, input().split()))
input = []

for _ in range(edge_cnt):
  line = sys.stdin.readline()
  input.append(line.split())

adj_list = defaultdict(list)
for u, v in input:
  adj_list[int(u)].append(int(v))
  adj_list[int(v)].append(int(u))

for key in adj_list:
  adj_list[key].sort()

dfs_visited = set()
dfs_result = []

bfs_visitied = set()
bfs_result = []

def dfs(start):
  dfs_visited.add(start)
  dfs_result.append(start)

  for node in adj_list[start]:
    if node not in dfs_visited:
      dfs(node)

def bfs(start):
  queue = deque([start])
  bfs_visitied.add(start)
  bfs_result.append(start)

  while queue:
    node = queue.popleft()

    for n in adj_list[node]:
      if n not in bfs_visitied:
        queue.append(n)
        bfs_visitied.add(n)
        bfs_result.append(n)

dfs(start)
bfs(start)
print(*dfs_result)
print(*bfs_result)